solarforecastarbiter/reports/figures.py

```diff
@@ -136,34 +162,37 @@ def timeseries(timeseries_cds, start, end, units, timezone='UTC'):
         name='timeseries')
 
     plotted_objects = 0
-    for obs_hash in np.unique(timeseries_cds.data['observation_hash']):
-        first_row = np.argwhere(
-            timeseries_cds.data['observation_hash'] == obs_hash)[0][0]
-        view = CDSView(source=timeseries_cds, filters=[
-            GroupFilter(column_name='observation_hash', group=obs_hash)
+    for obs_hash in np.unique(timeseries_meta_cds.data['observation_hash']):
+        metadata = _extract_metadata_from_cds(
+            timeseries_meta_cds, obs_hash, 'observation_hash')
+        pair_indices = _boolean_filter_indices_by_pair(
+            timeseries_value_cds, metadata['pair_index'])
+        view = CDSView(source=timeseries_value_cds, filters=[
+            BooleanFilter(pair_indices)
```

Comment:
Does a GroupFilter on pair_index not work? seems like that would allow us to skip making the boolean array